modules/contact/src/auxkernels/MortarFrictionalPressureVectorAux.C

```diff
@@ -0,0 +1,124 @@
+    mooseError(
+        "MortarFrictionalPressureVector auxiliary kernel can only be used with nodal kernels.");
+
+  // Kernel need to be boundary restricted
+  if (!this->_bnd)
+    paramError("boundary",
+               "MortarFrictionalPressureVector auxiliary kernel must be restricted to a boundary.");
+
+  // Get mortar interfaces
+  const auto & displaced_mortar_interfaces = _fe_problem.getMortarInterfaces(/*displaced=*/true);
+
+  if (displaced_mortar_interfaces.size() == 0)
+    paramError("tangent_one",
+               "No mortar interface could be identified in this problem. Make sure mortar contact "
+               "is enabled.");
+
+  const auto mortar_dimension = _fe_problem.mesh().dimension() - 1;
+  if (mortar_dimension == 1)
+    paramError("tangent_two",
+               "MortarFrictionalPressureVector auxiliary kernel can only be used in "
+               "three-dimensional problems");
+}
+
+Real
+MortarFrictionalPressureVectorAux::computeValue()
+{
+  const auto & displaced_mortar_interfaces = _fe_problem.getMortarInterfaces(/*displaced=*/true);
+
+  // Get automatic generation object for the boundary pair this auxiliary acts on.
+  if (displaced_mortar_interfaces.count(std::make_pair(_primary_id, _secondary_id)) != 1)
+    mooseError("primary_boundary",
+               "The boundary pairs do not correspond to a single mortar contact boundary pair. "
+               "Please revise your input file for proper mortar contact constraints and mortar "
+               "frictional pressure vector auxiliary variable definition.");
+
+  const AutomaticMortarGeneration & mortar_generation_object =
+      displaced_mortar_interfaces.at(std::make_pair(_primary_id, _secondary_id));
+
+  // Note: Nodal kernels cannot make use of _current_elem
+  const std::map<dof_id_type, std::vector<dof_id_type>> & node_to_elem_map = _mesh.nodeToElemMap();
+  auto node_to_elem_pair = node_to_elem_map.find(_current_node->id());
+  mooseAssert(node_to_elem_pair != node_to_elem_map.end(), "Missing entry in node to elem map");
+  std::vector<dof_id_type> element_ids = node_to_elem_pair->second;
+
+  // We can pick any element id since we are looping over nodes below.
+  const Elem * lower_dimensional_element =
+      mortar_generation_object.getSecondaryLowerdElemFromSecondaryElem(element_ids[0]);
+
+  // Get nodal tangents for this element
+  std::array<std::vector<Point>, 2> nodal_tangents =
+      mortar_generation_object.getNodalTangents(*lower_dimensional_element);
+
+  Real tangent_one_component = 0;
+  Real tangent_two_component = 0;
+  //
+  for (const auto lowerd_node : make_range(lower_dimensional_element->n_nodes()))
+  {
+    if (_fe_problem.mesh().getMesh().node_ptr(_current_node->id()) ==
+        _fe_problem.mesh().getMesh().node_ptr(lower_dimensional_element->node_id(lowerd_node)))
+    {
+      tangent_one_component = nodal_tangents[0][0](_component);
+      tangent_two_component = nodal_tangents[1][0](_component);
+      break;
+    }
+  }
```

Comment:
I don't really understand what's going on here. I feel like the second index of `nodal_tangents` should be with `lowerd_node` as opposed to always 0?

Reply:
I don't think we can be sure that `element_ids[0]` zeroth node will always be the node that corresponds to `_current_node`

Reply:
> I don't really understand what's going on here. I feel like the second index of nodal_tangents should be with lowerd_node as opposed to always 0?

Yeah, you are right. I was testing this on a flat surface, so couldn't see the error there. Initially, I didn't have the logic and hard coded it to zero. When I put the logic together I forgot to actually use it.  I believe this update should also address your second comment.